report/CommunicationStrategies/combining.tex

```diff
@@ -0,0 +1,33 @@
+The first being, which "screens" are there in the app, by screens we refere to a collection of Graphical User Interaction (GUI) Elements all accessible in a perticular state of the application, and which GUI elements are present on said screens.
+The second thing is how does those screens and GUI elements behave, when using the application.
+
+Out proposal is then a software that allow the customer to set requirements for the system in a What You See Is What You Get (wysiwyg) envoriment where the customer can create screen, or more generalized a GUI-component, specify which GUI elements (such as Text, Image, Inputfield, Button, etc.) should be present on said screen. This can be translated into "When showing [name]-screen, Then I should see [list of elements]", the only thing missing for being a contract is the pre-condition. For the pre condition we suggest two repositories, an actor repository where the customer can create actors for the system and a entity repository where the customer can create data-structures that they already know about with out the system. To illustrate what a data structure that customer already knows about could be, lets consider a Library, let say that we are developing a system for a Library, the customer then already know that they have books and that a book have a title, a author, etc. All these data points are well known for the customer in their context. So the custoemr can go into the Entity repository create a entity called Book and specify that a book has a title, a author, etc. But the customer cannot specify technical details such as what data type is the title. When creating a screen the customer could then specify given a book with title "Harry Potter" and as a citizen, when i go to the book-detail-scren, i should see a Text element with the data "Harry Potter". This can then be translated into "Given a Book{title: "Harry Potter"} and as a Citizen, When i go to book-details-screen, I should see "Harry Potter"". This could then be converted into executable tests, an example in psydo-code could look like:
+
+\begin{lstlisting}
+    void test_detail_screen(TestHelper Helper){
+        int BookId = Helper->factory->book(title: "Harry Potter")->create();
+
+        Helper->ActAs("Citizen");
+
+        Screen detailScreen = new DetailScreen(BookId);
+
+        assertContains(detailScreen, "Harry Potter");
+    }
+\end{lstlisting}
+
+
+Finally after a customer have specified the requirements for a screen, the screen can now act as preconditions for the more usecase driven requirements. An example of this could be in the wysiwyg envoriment the customer can create a new usecase and specify that it is a usecase for the login screen, then create a event list like: 1) set the Username field to empty 2) click login, and then specify that an error should be shown to the user. This will also be translated into to Hoare Logic, which then can be translated into executable tests.
```

Comment:
```suggestion
Finally after a customer have specified the requirements for a screen, the screen can now act as preconditions for the more use case driven requirements. 
An example of this could be in the wysiwyg environment the customer can create a new use case and specify that it is a use case for the login screen, then create a event list like: 
1) set the Username field to empty 2) click login, and then specify that an error should be shown to the user. This will also be translated into to Hoare Logic, which then can be translated into executable tests.
```